File: src/main/java/com/pauljoda/modularsystems/core/items/BaseItem.java
package com.pauljoda.modularsystems.core.items;

import com.pauljoda.modularsystems.core.ModularSystems;
import com.pauljoda.modularsystems.core.lib.Reference;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.item.Item;

public class BaseItem extends Item {

    private String name;

    public BaseItem(String name, int maxStackSize) {
        this.setCreativeTab(ModularSystems.tabModularSystems);
        this.maxStackSize = maxStackSize;
        this.setUnlocalizedName(Reference.MOD_ID + ":" + name);
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public void registerIcons(IIconRegister register) {
        this.itemIcon = register.registerIcon(Reference.MOD_ID + ":" + name);
    }
}

File: src/main/scala/com/teambr/modularsystems/core/common/items/BaseItem.scala
package com.teambr.modularsystems.core.common.items

import com.teambr.modularsystems.core.ModularSystems
import com.teambr.modularsystems.core.lib.Reference
import net.minecraft.item.Item

/**
 * Modular-Systems
 * Created by Dyonovan on 05/08/15
 */
class BaseItem(name: String, maxStackSize: Int) extends Item {

    setCreativeTab(ModularSystems.tabModularSystems)
    setMaxStackSize(maxStackSize)
    setUnlocalizedName(Reference.MOD_ID + ":" + name)

    def getName: String = { name }

}
